Entity/Player/Player.gd

```diff
@@ -2,11 +2,50 @@ extends Entity
 class_name Player
 ## Base player Entity class.
 
+var _should_save_persistent_data: bool = true
+
 # @Override
 func _ready() -> void:
 	super()
 	PlayerManager.set_player(self)
+	
+	_load_persistent_data()
 
 
 func _exit_tree() -> void:
+	# Save player data when they are destroyed (combat end)
 	PlayerManager.set_player(null)
+	
+	if _should_save_persistent_data:
+		_save_persistent_data()
```

Comment:
shouldn't the save happen before setting the player to null ?

Reply:
It doesn't really matter since all PlayerManager.set_player does is remove the reference in PlayerManager